client/signals/handlers.py

```diff
@@ -0,0 +1,134 @@
+        """
+        On login `last_login` field change for ignoring this change we used this check.
+        We are expecting last_login will never changed from django admin panel.
+        """
+        user_data = {
+            'fullname': '{} {}'.format(instance.first_name, instance.last_name)
+        }
+        task_sync_user_profile_info_with_nodebb.delay(username=instance.username, **user_data)
+
+
+@receiver(post_save, sender=UserProfile)
+def sync_user_profile_info_with_nodebb(sender, instance, **kwargs):
+    """
+    If some changed occurs in the User Profile, makes sure that
+    these changes are also made at Nodebb side.
+
+    Args:
+        sender (str): Name of Sender model
+        instance: Newly created entry of Model.
+        **kwargs:  All remaining fields.
+    """
+    user = instance.user
+    user_data = {
+        'fullname': instance.name,
+        'location': '{}, {}'.format(
+            instance.city, instance.country.name),
+        'birthday': '01/01/%s' % instance.year_of_birth
+    }
+    task_sync_user_profile_info_with_nodebb.delay(username=user.username, **user_data)
+
+
+@receiver(pre_delete, sender=User)
+def delete_user_from_nodebb(sender, instance, **kwargs):
+    """
+    Deletes the user from NodeBB if it is deleted from edX.
+
+    Args:
+        sender (str): Name of Sender model
+        instance: Newly created entry of Model.
+        **kwargs:  All remaining fields.
+    """
+    task_delete_user_from_nodebb.delay(username=instance.username)
+
+
+@receiver(post_save, sender=CourseOverview)
+def create_category_on_nodebb(sender, instance, created, update_fields, **kwargs):
+    """
+    Whenever a new course is created in edX, creates a new category in NodeBB.
+
+    Args:
+        sender (str): Name of Sender model
+        instance: Newly created entry of Model.
+        created: Flag will be true if new instance created or false if upgraded.
+        update_fields:  Contain frozen_set of the name of fields updated.
+        **kwargs:  All remaining fields.
+    """
+    if created:
+        course_data = {
+            'name': '{}-{}-{}-{}'.format(instance.display_name, instance.id.org, instance.id.course, instance.id.run),
+        }
+        task_create_category_on_nodebb.delay(course_id=instance.id, course_display_name=instance.display_name,
+                                             **course_data)
+
+
+@receiver(pre_delete, sender=EdxNodeBBCategory)
+def delete_category_from_nodebb(sender, instance, **kwargs):
+    """
+    Deletes the category from NodeBB if it is deleted from edX.
+
+    Args:
+        sender (str): Name of Sender model
+        instance: Newly created entry of Model.
+        **kwargs:  All remaining fields.
+    """
+    category_id = instance.nodebb_cid
+    task_delete_category_from_nodebb.delay(category_id)
+
+
+@receiver(post_save, sender=CourseEnrollment)
+def manage_membership_on_nodebb_group(sender, instance, **kwargs):
```

Comment:
`manage_` is ambiguous. Use a proper verb like `add`

Reply:
This function is managing both the join and un_join operation based on user enrollment in the course. That's why we name it to manage.